data_model/src/permission/validator.rs

```diff
@@ -142,8 +39,6 @@ model! {
         Instruction(Instruction),
         /// [`QueryBox`] execution operations
         Query(QueryBox),
-        /// [`Expression`] evaluation operation
-        Expression(Expression),
```

Comment:
> this never made sense, did it?

I agree, looks like there was no meaning in it

> it was needed only so that wasm smart contract can communicate with the host environment

Yes, validator still can receive any of this three variants as input, so we need to keep `NeedsPermissionBox`